src/cascadia/TerminalApp/AppCommandlineArgs.cpp

```diff
@@ -0,0 +1,605 @@
+    // command was parsed.
+    _newTabCommand.subcommand->callback([&, this]() {
+        // Buld the NewTab action from the values we've parsed on the commandline.
+        auto newTabAction = winrt::make_self<implementation::ActionAndArgs>();
+        newTabAction->Action(ShortcutAction::NewTab);
+        auto args = winrt::make_self<implementation::NewTabArgs>();
+        // _getNewTerminalArgs MUST be called before parsing any other options,
+        // as it might clear those options while finding the commandline
+        args->TerminalArgs(_getNewTerminalArgs(_newTabCommand));
+        newTabAction->Args(*args);
+        _startupActions.push_back(*newTabAction);
+    });
+}
+
+// Method Description:
+// - Adds the `split-pane` subcommand and related options to the commandline parser.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void AppCommandlineArgs::_buildSplitPaneParser()
+{
+    _newPaneCommand.subcommand = _app.add_subcommand("split-pane", NEEDS_LOC("Create a new pane"));
+    _addNewTerminalArgs(_newPaneCommand);
+    _horizontalOption = _newPaneCommand.subcommand->add_flag("-H,--horizontal",
+                                                             _splitHorizontal,
+                                                             NEEDS_LOC("Create the new pane as a horizontal split (think [-])"));
+    _verticalOption = _newPaneCommand.subcommand->add_flag("-V,--vertical",
+                                                           _splitVertical,
+                                                           NEEDS_LOC("Create the new pane as a vertical split (think [|])"));
+    _verticalOption->excludes(_horizontalOption);
+
+    // When ParseCommand is called, if this subcommand was provided, this
+    // callback function will be triggered on the same thread. We can be sure
+    // that `this` will still be safe - this function just lets us know this
+    // command was parsed.
+    _newPaneCommand.subcommand->callback([&, this]() {
+        // Buld the SplitPane action from the values we've parsed on the commandline.
+        auto splitPaneActionAndArgs = winrt::make_self<implementation::ActionAndArgs>();
+        splitPaneActionAndArgs->Action(ShortcutAction::SplitPane);
+        auto args = winrt::make_self<implementation::SplitPaneArgs>();
+        // _getNewTerminalArgs MUST be called before parsing any other options,
+        // as it might clear those options while finding the commandline
+        args->TerminalArgs(_getNewTerminalArgs(_newPaneCommand));
+        // Make sure to use the `Option`s here to check if they were set -
+        // _getNewTerminalArgs might reset them while parsing a commandline
+        if ((*_horizontalOption || *_verticalOption) && (_splitHorizontal))
+        {
+            args->SplitStyle(SplitState::Horizontal);
+        }
+        else
+        {
+            args->SplitStyle(SplitState::Vertical);
+        }

```

Comment:
Shouldn't we support `Auto` too? If this is for later, I guess could you add an issue here plz